hoomd/md/TwoStepRATTLEBD.h

```diff
@@ -0,0 +1,476 @@
+
+        //! Get the number of degrees of freedom granted to a given group
+        virtual Scalar getTranslationalDOF(std::shared_ptr<ParticleGroup> group)
+        {
+        // get the size of the intersection between query_group and m_group
+        unsigned int intersect_size = ParticleGroup::groupIntersection(group, m_group)->getNumMembersGlobal();
+
+        return ( m_sysdef->getNDimensions() - 1 ) * intersect_size;
+        }
+
+        /// Sets eta
+        void setEta(pybind11::object eta){ m_eta = pybind11::cast<Scalar>(eta); };
+
+        /// Gets alpha
+        pybind11::object getEta(){ return pybind11::cast(m_eta); };
+
+    protected:
+        Manifold m_manifold;  //!< The manifold used for the RATTLE constraint
+        bool m_noiseless_t;
+        bool m_noiseless_r;
+        Scalar m_eta;                      //!< The eta value of the RATTLE algorithm, setting the tolerance to the manifold
+
+        GPUArray<Scalar3> m_f_brownian; //! Brownian random force
+    };
+
+
+
+/*! \file TwoStepRATTLEBD.h
+    \brief Contains code for the TwoStepRATTLEBD class
+    \Warning NDOF is still 3*(N_part-1) and not 2*(N_part-1)!!! Has to be considered in thermodynamic quantities calculations.
+*/
+
+/*! \param sysdef SystemDefinition this method will act on. Must not be NULL.
+    \param group The group of particles this integration method is to work on
+    \param manifold The manifold describing the constraint during the RATTLE integration method
+    \param T Temperature set point as a function of time
+    \param seed Random seed to use in generating random numbers
+    \param use_alpha If true, gamma=alpha*diameter, otherwise use a per-type gamma via setGamma()
+    \param alpha Scale factor to convert diameter to gamma
+    \param noiseless_t If set true, there will be no translational noise (random force)
+    \param noiseless_r If set true, there will be no rotational noise (random torque)
+    \param eta Tolerance for the RATTLE iteration algorithm
+*/
+template< class Manifold>
+TwoStepRATTLEBD<Manifold>::TwoStepRATTLEBD(std::shared_ptr<SystemDefinition> sysdef,
+                           std::shared_ptr<ParticleGroup> group,
+                           Manifold manifold,
+                           std::shared_ptr<Variant> T,
+                           unsigned int seed,
+                           Scalar eta
+                           )
+  : TwoStepLangevinBase(sysdef, group, T, seed), m_manifold(manifold),
+    m_noiseless_t(false), m_noiseless_r(false), m_eta(eta)
+    {
+    m_exec_conf->msg->notice(5) << "Constructing TwoStepRATTLEBD" << endl;
+
+    bool manifold_error = m_manifold.validate(m_pdata->getBox());
+
+    if( manifold_error){
+        this->m_exec_conf->msg->error() << "manifold." << Manifold::getName() << ": Parts of the manifold are outside the box. Constrained particle positions are probably incorrect"
+                 << std::endl;
+        throw std::runtime_error("Error setting Manifold class");
```

Comment:
```suggestion
        throw std::runtime_error("Parts of the manifold are outside the box");
```
Ah, I see, this is where `getName` is used. We can remove the error message and replace with a descriptive exception. The exception traceback will include the class name details the users needs to know.